### PR TITLE
#472 add UriGraphType

### PR DESCRIPTION
**Pull request by MNie**
Add support to urigraphtype
Fixes: #472

CC: @joemcbride

### CODE REVIEW

**Comment by joemcbride** [comment]
Hey, thanks for the PR though this is one of the cases where I'm not sure I want to add this since it is not defined in the core GraphQL spec.

For a "proper" implementation you would need to handle the `UriValue` in the `AstPrinter`. Also at present I do not believe you would ever get a `UriValue`, since it would get converted to a `StringValue` first.  To ever get a `UriValue` you would need to [alter this code](https://github.com/graphql-dotnet/graphql-dotnet/blob/bef9e5046ee532328609b2a86486e5cfc89be4aa/src/GraphQL/GraphQLExtensions.cs#L372).

File: src/GraphQL/ValueConverter.cs [hunk 1]
```diff
@@ -33,6 +33,9 @@ static ValueConverter()
             Register(typeof(long), typeof(int), LongToInt);
 
             Register(typeof(double), typeof(decimal), DoubleToDecimal);
+
+            Register(typeof(string), typeof(Uri), ParseUri);
+            Register(typeof(Uri), typeof(Uri), ParseUri);
```
Comment (joemcbride) [inline src/GraphQL/ValueConverter.cs]: Don’t need Uri->Uri registration.  The code checks to see if the value is of the requested type and returns if so.

Comment (MNie) [reply src/GraphQL/ValueConverter.cs]: fix'd

**Comment by MNie** [comment]
@joemcbride ok I get it :). I fix the pr, so now all should be fine. Since I used this library in my daily work I want to contribute to it somehow so I grab some issues and try to resolve/fix them. But as you said if the URI is not available in GraphQL. I'm not 100% sure that it would be good to add it to a GraphQL-Dotnet? But on the other hand, thanks to it we have a confidence that field of type URI would be properly formatted no matters what is done on the server side.

**Pull request merged**